### PR TITLE
Components: upgrade Ariakit to latest

### PR DESCRIPTION
**Pull request by DaniGuardiola**

### CODE REVIEW

**Comment by DaniGuardiola** [comment]
@mirka @tyxla can you help me figure out what I need to look into to make sure this upgrade can safely ship? I don't wanna miss anything.

Linking this comment where some of this was already discussed: https://github.com/WordPress/gutenberg/pull/60897#discussion_r1572453947

**Comment by mirka** [comment]
What we usually do is:

- Check the Ariakit changelog for things that might affect us, and smoke test any affected components.
- Address any test failures.

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
 		"IS_GUTENBERG_PLUGIN": true
 	},
 	"dependencies": {
+		"@ariakit/react": "0.4.5",
```
Comment (mirka) [inline package.json]: Note that the dep is declared in the components package, not in the root package.json.

**Comment by tyxla** [comment]
> What we usually do is:
> 
> * Check the Ariakit changelog for things that might affect us, and smoke test any affected components.
> * Address any test failures.

What Lena said. The ariakit changelog is usually pretty thorough. I'd devote special attention to the breaking changes there, although there might be something unintentionally breaking. 

As we also discussed earlier today, many of our components don't use Ariakit, so the impact of the update won't be as if every component uses Ariakit under the hood.

File: package.json (hunk 1 above)
Comment (tyxla) [inline package.json]: Also we might want to update `@ariakit/test` together with `@ariakit/react`.

Comment (DaniGuardiola) [reply package.json]: Fixed, thanks for the heads up.

Comment (DaniGuardiola) [reply package.json]: Makes sense, done 👍

**Comment by diegohaz** [comment]
Looks like most of the failing tests are due to a change on `data-*` attributes: [Public data attributes have now boolean values](https://ariakit.org/changelog#public-data-attributes-have-now-boolean-values)

Also, tooltips no longer assign `aria-labelledby`/`aria-describedby` attributes: https://github.com/ariakit/ariakit/pull/3318

Other failures appear to prompt `act` warnings, potentially indicating an internal change in sync/async stuff. Typically, replacing `getBy` with `findBy`, or `waitFor` a change on the screen before making an assertion resolves the problem. Alternatively, you can use `@ariakit/test` events, which may also be beneficial and prevent these warnings from recurring since it doesn't depend on `act`. More context: https://github.com/WordPress/gutenberg/pull/54225#issuecomment-1711949971

**Comment by DaniGuardiola** [comment]
Thank you very much @diegohaz :)

I had parked this PR for a bit to focus on other stuff but your insights are gonna be super helpful as I come back to it (soon).

**Comment by ciampo** [comment]
hey @DaniGuardiola , do you think you'll be able to resume work on this? It would be good to have the latest version while wrapping up the rewrite of `CustomSelectControl` and `DropdownMenu`

**Comment by DaniGuardiola** [comment]
@ciampo alright, that's a great reason to bump this in my priority list 👍

**Comment by DaniGuardiola** [comment]
Moved to main repo, see https://github.com/WordPress/gutenberg/pull/62947

**Pull request closed without merging**